user_query.sql

```diff
@@ -0,0 +1,49 @@
+WITH observations_agg AS (
+    SELECT
+        t.user_id,
+        count(*) as obs_count,
+        sumMap(usage_details) as sum_usage_details,
+        sum(total_cost) as sum_total_cost
+    FROM
+        observations o
+            join traces t on t.id = o.trace_id
+            and t.project_id = o.project_id
+    WHERE
+        o.project_id = {projectId: String }
+      AND o.trace_id in (
+        SELECT
+        id
+        from
+        traces
+        where
+        user_id IN ({userIds: Array(String) })
```

Comment:
Does it make sense to include projectId here as well? even if we have a bloom filter, using parts of the order key should reduce the result set even further. Also, we avoid duplicates if userIds are not unique across projects.

Reply:
Also, you may want to include the IS NOT NULL and != '' filter here in case they bring any advantages. I would actually propose that we drop them in both queries as I don't see why they would help if we have the `IN` condition anyway.

Reply:
agree on all